### ISSUE TITLE
Connected to peer failed 

### ISSUE THREAD

**Issue by MrDarkSkil** [jeremija/peer-calls#27]
I tryed to install peer call to my server, it's work but when i join a call, peer connection fail :/

Look at bellow picture

![Image of peer fail](https://i.gyazo.com/1f0b1c40853889d03a971e825f207ff9.jpg)

**Comment by jeremija** [comment]
Unfortunately I don't think I can help you with this - does it work when you use peercalls.com? I noticed that sometimes corporate firewalls block connections. For example, when I'm at work I can establish a connection between my phone and a PC if they're on the same network, however it doesn't work when I try to reach out to somebody outside of the network, whereas if I'm at home everything works.

**Comment by MrDarkSkil** [comment]
With peercalls.com it's work.
But locally peer to peer connexion failed :/

I probably forget a file to configure the HPPS, can you tel me what file i need to edit for configure the HTTPS.

I edited /src/server/app.js file for configure the HTTPS.

**Comment by matikbird** [comment]
What did you edit in that file, MrDarkSkil?
Could you share the code?

**Comment by MrDarkSkil** [comment]
File: src/server/app.js


<pre>
#!/usr/bin/env node
'use strict';
var express = require('express');
var handleSocket = require('./socket.js');
var os = require('os');
var path = require('path');
var https = require('https');
var fs = require('fs');


// Require in express.Router Middleware.
var callRouter = require('./routes/call');
var siteRouter = require('./routes/index');

var app = express();


var hskey = fs.readFileSync('./ssl/myssl_key.key');
var hscert = fs.readFileSync('./ssl/myssl_cer.cer');
var ca = fs.readFileSync('./ssl/myssl_cer_INTERMEDIATE.cer');


var options = {
    ca: ca,
    key: hskey,
    cert: hscert
};

var server = https.createServer(options, app);

var http = server;
var io = require('socket.io')(http);

app.set('view engine', 'jade');
app.set('views', path.join(__dirname, '../views'));

app.use('/res', express.static(path.join(__dirname, '../res')));

if (__dirname.indexOf('/dist/') >= 0 || __dirname.indexOf('\\dist\\') >= 0) {
  app.use('/js', express.static(path.join(__dirname, '../client')));
  app.use('/css', express.static(path.join(__dirname, '../css')));
} else {
  const browserify = require('browserify-middleware');
  const less = require('less-middleware');
  browserify.settings({
    transform: ['babelify']
  });

  const tempDir = path.join(os.tmpDir(), 'node-peer-calls-cache');
  app.use('/js', browserify(path.join(__dirname, '../client')));
  app.use('/css', less(path.join(__dirname, '../less'), { dest: tempDir}));
  app.use('/css', express.static(tempDir));
  app.use('/css/fonts', express.static(
    path.join(__dirname, '../less/fonts')));
}

// using Express.Router Middleware
app.use('/call', callRouter);
app.use('/', siteRouter);



io.on('connection', socket => handleSocket(socket, io));

module.exports = { http, app };

</pre>

**Comment by matikbird** [comment]
I'm running into the same problem.
I had set up an HTTPS connection,
but the call always seems lost.

In peer-calls.com works. I can add as many tabs-browser as I want, and all of them work with my ugly face showing.
In my server, only I see myself; but the rest of myselves (!) are nowhere. They got lost in translation or something.
Look here: http://mrkapl.science:3000

**Comment by MrDarkSkil** [comment]
@jeremija can you give the peer-call.com source (without ssl file of course) ?

**Comment by matikbird** [comment]
As you can see, I have two instances running in the same call; and yet, I only see myself.

![screenshot from 2016-11-20 15 29 04](https://cloud.githubusercontent.com/assets/12424545/20465184/2c704f96-af36-11e6-8e6b-466ea1ecc965.png)

**Comment by matikbird** [comment]
But, if you look at peercalls.com, it totally works!
Why can this be happening, @jeremija?
Maybe you missed something in the last branch that you put into peer-calls.

By the way, amazing work. Really clever.
![screenshot from 2016-11-20 15 31 07](https://cloud.githubusercontent.com/assets/12424545/20465198/71aa29ce-af36-11e6-8cb1-59c8cd8a7993.png)

**Comment by matikbird** [comment]
PD: The console error throws no errors, all calls made are working perfectly.
Why would this happen?
In my localhost, and my server, the same thing.
Maybe it's something in the core, @jeremija

**Comment by matikbird** [comment]
But yes, @jeremija 
Perhaps sharing the code of peercalls.com (without the SSL, like Mr said), and we test it out! (:

**Comment by matikbird** [comment]
@MrDarkSkil he already did, per haps we are doing something wrong. or maybe something went wrong in the last merge.

PD100: I totally disabled my firewall and all, and the same..

**Comment by jeremija** [comment]
To enable https in node natively, you'd have to use the `https` module similar in the manner mrdarkskill described. Check the https part int he docs [here][1].

In production I use nginx as a reverse proxy server for my peer calls. The app config you'd need looks something like this:

```nginx
server {
        server_tokens off;

        listen                  443;
        server_name             yourdomain.com;

        ssl                     on;
        ssl_certificate         /path/to/fullchain.pem;
        ssl_certificate_key     /path/to/privkey.pem;

        # optinally, enable strong ciphers to make the connection more secure
        # ssl_protocols           TLSv1.2;
        # ssl_ciphers             ...;

        location / {
                proxy_set_header        X-Real-IP       $remote_addr;
                proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header        Host            $http_host;

                # the following enables websockets
                proxy_http_version      1.1;
                proxy_set_header Upgrade        $http_upgrade;
                proxy_set_header Connection     "upgrade";

                proxy_pass      http://127.0.0.1:3000;
        }

}
```

[1]: http://expressjs.com/en/api.html#app.listen

**Comment by matikbird** [comment]
I did.
I have the PeerCalls with HTTPS up and running, @jeremija : https://mrkapl.science:3000

Check it out yourself. Open more than one tab in the same link.
Why? :(

**Comment by matikbird** [comment]
Not what I meant, @jeremija 
but thanks!

What I meant it's that only ONE can see how many users are connected to the peer; but only shows the video of myself. Not others.
Try it out yourself, above I left the link

**Comment by matikbird** [comment]
```

  let peer = peers[user.id] = Peer.init({
    initiator: '/#' + socket.id === initiator,
    stream,
    config: {
      iceServers: [{
        url: 'stun:23.21.150.121',
        urls: 'stun:23.21.150.121'
      }, {
        url: 'turn:numb.viagenie.ca',
        urls: 'turn:numb.viagenie.ca',
        credential: 'muazkh',
        username: 'webrtc@live.com'
      }]
    }
  });
```

What does this means, @jeremija ?
What are those IP's?
Tried to access them but refused me the attempt.

**Comment by MrDarkSkil** [comment]
I installed ngnix and i configured https with your ngnix config ans he does't work 😠 

I thinks to make https://simplewebrtc.com/ to integrate to my website because your system is so broken

**Comment by matikbird** [comment]
Then fix it instead of being a bitch about it.
& to me, it seems far from broken when I go to peercalls.com

**Comment by matikbird** [comment]
```
 let peer = peers[user.id] = Peer.init({
    initiator: '/#' + socket.id === initiator,
    stream,
    config: {
      iceServers: [{
        url: 'stun:23.21.150.121',
        urls: 'stun:23.21.150.121'
      }, {
        url: 'turn:numb.viagenie.ca',
        urls: 'turn:numb.viagenie.ca',
        credential: 'muazkh',
        username: 'webrtc@live.com'
      }]
    }
  });
```

Maybe this could be the mistake?
What do you think, @jeremija ?
http://mrkapl.science:3000

**Comment by jeremija** [comment]
@MrDarkSkil if you don't like this project, feel free to switch whatever whatever you feel competent using. i spent my free time to develop and it works for me, and for many other people; so you can either figure out what the problem is on your end and submit a pull request to fix it if you find out the code is to blame, or stop making inarticulate complaints.

@matikbird this is a list of stun and turn servers used to establish a peer-to-peer connection. you can read more about it [here][1]. normally i would advise against storing credentials in version control, however these credentials are publicly available, and will be available from the client side anyway so it doesn't really make sense to hide them.

the bottom line is, i cannot, nor do i have time to provide free support for your personal servers, so i'm closing this.

[1]: https://tools.ietf.org/html/rfc5766
